bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/profile/ProfileTypeResource.java

```diff
@@ -151,31 +144,26 @@ private boolean profileTypeMatchesItemType(ProfileType pt, String itemType) {
                 || supportedItemTypesOnProfileType.contains(itemType);
     }
 
-    private boolean triggerProfileMatchesProfileType(ProfileType profileType, ChannelType channelType) {
-        if (profileType instanceof TriggerProfileType triggerProfileType) {
-            if (triggerProfileType.getSupportedChannelTypeUIDs().isEmpty()) {
-                return true;
-            }
+    private boolean profileTypeMatchesChannelType(ProfileType profileType, ChannelType channelType) {
+        @Nullable
```

Comment:
You don't need `@Nullable` annotation for local variables.

Reply:
My IDE (VSCode) disagrees and outputs a warning saying that the value cannot be null. That's what I've been doing in my plugin bundles, but I can remove it if you prefer.

Reply:
IMO it's wrong. The default locations for `@NonNullByDefault` are `PARAMETER, RETURN_TYPE, FIELD, TYPE_BOUND, TYPE_ARGUMENT`, so it does not apply to local variables.

Reply:
Fair enough, I have removed them